lib/cors/validate.js

```diff
@@ -15,23 +17,34 @@ function validateCors (
   if (!cors) {
     return Promise.reject(new Error('Must specify cors configuration' + HELP))
   }
+  // Origins
   if (!cors.origins || !Array.isArray(cors.origins) || !cors.origins.length) {
     return Promise.reject(new Error('Must specify at least a single allowable origin in cors configuration' + HELP))
   }
-  if (cors.origins.some((origin) => origin === '*')) {
-    return Promise.reject(new Error('Cannot specify all origins as allowable in cors configuration i.e. "*"' + HELP))
-  }
   const invalidOrigins = cors.origins.reduce((invalidOrigins, origin) => {
-    if (!validUrl.isWebUri(origin)) {
+    if (origin !== '*' && !validUrl.isWebUri(origin)) {
       invalidOrigins.push(origin)
     }
     return invalidOrigins
   }, [])
   if (invalidOrigins.length) {
     return Promise.reject(new Error('The following origins in cors configuration are not valid: ' + invalidOrigins.join(', ')))
   }
+  // Headers
   if (cors.headers && !Array.isArray(cors.headers)) {
-    return Promise.reject(new Error('Invalid headers in cors configuration' + HELP))
+    return Promise.reject(new Error('Headers must be an array of strings in cors configuration' + HELP))
+  }
+  // Exposed Headers
+  if (cors.exposedHeaders && !Array.isArray(cors.exposedHeaders)) {
+    return Promise.reject(new Error('Exposed headers must be an array of strings in cors configuration' + HELP))
+  }
+  // Max Age
+  if (!isNumber(cors.maxAge)) {
+    return Promise.reject(new Error('Max age must be a number in cors configuration' + HELP))
+  }
+  // Credentials
+  if (!isBoolean(cors.credentials)) {
```

Comment:
`lodash.isBoolean(value)` makes sense if we're doing it in a bunch of places. But if this is the only occurrence, then maybe consider good old `typeof value === 'boolean'`?